### PR TITLE
Add logger (WIP)

### PR DESCRIPTION
**Pull request by JakeRoggenbuck**

### CODE REVIEW

File: src/main/util.hpp [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+enum Severity {
+    NORMAL,
+    DEBUG,
+    ERROR,
+    WARNING,
+    FATAL,
+};
+
+const char *SeverityText[] = {
+    "", "Normal: ", "Debug: ", "Error: ", "Warning: ", "Fatal: "};
```
Comment (adamhutchings) [inline src/main/util.hpp:15]: Why is there an empty string at the beginning? Won't this make the `SeverityText[severity]` return the wrong message?

File: src/main/util.hpp [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+enum Severity {
+    NORMAL,
+    DEBUG,
+    ERROR,
+    WARNING,
+    FATAL,
+};
+
+const char *SeverityText[] = {
```
Comment (adamhutchings) [inline src/main/util.hpp:14]: This is minor, but because this is a variable could we call it `severity_text`?

**Review by adamhutchings — COMMENTED** [review]
Two minor changes.

File: src/main/util.hpp (hunk 1 above)
Comment (JakeRoggenbuck) [reply src/main/util.hpp:15]: Do enums start at 0 or 1 by default? Thinking about it more, 0 makes more sense. I can check

File: src/main/util.hpp (hunk 2 above)
Comment (JakeRoggenbuck) [reply src/main/util.hpp:14]: I would love to use snake case more than anything. `snake_case > *`

Comment (Sophon96) [reply src/main/util.hpp]: Its a const, so shouldn't it be `SCREAMING_SNAKE_CASE`?

**Pull request merged**